src/data/reducers/courseInfo.js

```diff
@@ -0,0 +1,105 @@
+import DiscoveryDataApiService from '../services/DiscoveryDataApiService';
+
+
+// *** CONSTANTS ***
+
+export const FAIL_COURSE_INFO = 'FAIL_COURSE_INFO';
+export const RECEIVE_COURSE_INFO = 'RECEIVE_COURSE_INFO';
+export const REQUEST_COURSE_INFO = 'REQUEST_COURSE_INFO';
+
+const UUID_REGEX = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;
+
+
+// *** ACTIONS ***
+
+export function failCourseInfo(id, error) {
+  return { type: FAIL_COURSE_INFO, id, error };
+}
+
+export function receiveCourseInfo(id, data) {
+  return { type: RECEIVE_COURSE_INFO, id, data };
+}
+
+export function requestCourseInfo(id) {
+  return { type: REQUEST_COURSE_INFO, id };
+}
+
+export function fetchCourseInfo(id) {
+  return (dispatch) => {
+    // We only support UUIDs right now, not course keys
+    if (!UUID_REGEX.test(id)) {
+      const error = `Could not get course information. ${id} is not a valid course UUID.`;
+      dispatch(failCourseInfo(id, error));
+      return Promise.resolve(); // early exit with empty promise
+    }
+
+    dispatch(requestCourseInfo(id));
+
+    return DiscoveryDataApiService.fetchCourse(id)
+      .then((response) => {
+        const course = response.data;
+
+        // Confirm it looks vaguely correct
+        if (!course || !('key' in course)) {
+          throw Error('Did not understand response.');
+        }
+
+        dispatch(receiveCourseInfo(id, course));
+      })
+      .catch((error) => {
+        dispatch(failCourseInfo(
+          id,
+          `Could not get course information. ${error.toString()}`,
+        ));
+      });
+  };
+}
+
+
+// *** REDUCER ***
+
+const initialState = {
+  data: {},
+  isFetching: false,
+  error: null,
+};
+
+export function oneCourseInfo(state = initialState, action) {
+  switch (action.type) {
+    case FAIL_COURSE_INFO:
+      return Object.assign({}, state, {
+        data: {},
+        isFetching: false,
+        error: action.error,
+      });
+    case RECEIVE_COURSE_INFO:
+      return Object.assign({}, state, {
+        data: action.data,
+        isFetching: false,
+        error: null,
+      });
+    case REQUEST_COURSE_INFO:
+      return Object.assign({}, state, {
+        data: {},
+        isFetching: true,
+        error: null,
+      });
+    default:
+      return state;
+  }
+}
+
+function courseInfo(state = {}, action) {
+  switch (action.type) {
+    case FAIL_COURSE_INFO:
+    case RECEIVE_COURSE_INFO:
+    case REQUEST_COURSE_INFO:
+      return Object.assign({}, state, {
+        [action.id]: oneCourseInfo(state[action.id], action),
+      });
+    default:
+      return state;
+  }
+}
+
+export default courseInfo;
```

Comment:
It might seem cumbersome at this point, but I believe there would be value in splitting the sections of this file into 3 directories and files like they have in edx-portal. 

* actions
* constants
* reducers

It may seem excessive right now, but it will set the tone for this work as it grows.  The separation of concerns will hopefully make it easier to navigate the code.

You thoughts @mikix ?

Reply:
I would agree if we were also grouping reducers. But it feels excessive to separate out reducer groups AND actions/constants. That is:

```
reducers.js
actions.js
constants.js
```

^ That seems reasonable. As does: 

```
reducers/courseInfo.js
reducers/courseTable.js
```

But the following just seems like we're getting in our own way:

```
actions/courseInfo.js
actions/courseTable.js
constants/courseInfo.js
constants/courseTable.js
reducers/courseInfo.js
reducers/courseTable.js
```

But I can disagree and commit with the best of them.